### DISCUSSION TITLE
v3.0.1

### DISCUSSION THREAD

**Discussion by osy** [osy86/HaC-Mini#577] [Releases]
**IMPORTANT READ THE FOLLOWING BEFORE UPDATING**: The new OpenCore has an incompatible change with v2.17 and below. As a result, if you run the update and restart, you will not be able to find the boot device. There are a few ways to complete the update:

* Option 1: install v3.0 on a USB drive with the "Show boot menu" option set. Then boot into the USB drive which loads OpenCore and select the Reset NVRAM option. Then you must boot from the USB again and select macOS and install v3.0 on your main partition.
* Option 2: install v3.0 on your main partition and *also* on a USB drive (with “Show boot menu” set). Reboot and select your USB drive. Select Reset NVRAM and remove the USB drive when it reboots.
* Option 3: if you've already installed v3.0 on your main partition without doing the above, you can get into UEFI Shell and type in `fs0:\EFI\OC\OpenCore.efi` from the shell prompt.
* Option 4: if all else fails, you can do a BIOS reset which should reset all settings as well as the boot options. Make sure to set up the BIOS settings again with your old settings!

## Changes (v3.0.1)

* Fixed an issue where Intel Wireless caused panic on boot

## Changes (v3.0)

* Add fans, voltage, and temperature sensors (#553)
* Translate to German (thanks @SkyrilHD)
* Intel wireless option now installs AirportItlwm instead of Itlwm

## Driver Updates (v3.0)

* AppleALC 1.5.6 => 1.6.0
* BrcmPatchRAM 2.5.5 => 2.5.8
* IntelMausi 1.0.5 => 1.0.6
* Lilu 1.5.0 => 1.5.3
* NVMeFix 1.0.5 => 1.0.7
* OpenCore 0.6.5 => 0.6.9
* VirtualSMC 1.1.9 => 1.2.3
* WhateverGreen 1.4.6 => 1.4.9

## Release Schedule

As this project reaches maturity, HaC mini will only be updated for bug fixes and major features. We will no longer be updating the driver KEXTs monthly.

We aim to have two major releases a year: one targeting a major update for macOS due to anticipated major changes (for example when macOS 12 comes out). A second one mid-way through the update cycle. Each major update will increment the second version number (for example v3.0 => v3.1). A major update will include any new features as well as bumping all drivers to the latest released versions.

When bug fixes are needed, we will only update the relevant drivers and the third version number will be incremented (for example v3.0 => v3.0.1).

Each major update will first be released as a pre-release as marked on Github. If some time passes and no major bugs are reported, then the pre-release will be promoted to the full release and will be the recommended build to install. Non-advanced users are recommended to only install full release builds as they will be more stable.

When reporting an issue, please make sure to include the version you are using.

<hr /><em>This discussion was created from the release <a href='https://github.com/osy/HaC-Mini/releases/tag/v3.0.1'>v3.0.1 (PRE-RELEASE)</a>.</em>

**Comment by Malumen** [comment]
Thank you for your work Osy!!!!!

**Reply by Malumen** [reply]
NOTE: I chose _Option 2_: installed on main SSD and also on a spare USB with `Install macOS Catalina` installed onto it. Rebooted -> no boot device found. Rebooted again, spammed `F10` to get to the boot picker menu (Intel-based, not the OpenCore one) and then selected the spare USB I have. Loaded OpenCore fine, then hit `Reset NVRam` and after the monitor went blank I pulled out the USB drive. After the system came back up, it pulled OpenCore from the main SSD successfully, I hit `Reset NVRam` one more time to be safe, and then rebooted into macOS.

Everything seems OK.

Note2: Osy seems to be using the `Force Kext` method to load AirportItlwm. Kudos to Osy for getting it working as when I was initially testing the kext I was forced to migrate to the `Enable SecureBootModel as Default, change ApECID to 0` method in the `config.plist`. (FYI, if anyone starts to get WiFi drop-outs or weird transmission speeds or the WiFi straight up does not load on boot, you'll need to enact the changes I mentioned above)

**Reply by Malumen** [reply]
@osy After applying latest macOS security update, the WiFi stopped working.

I have remedied this by changing some `config.plist` parameters:

`ApECID` to `0`
`SecureBootModel` to `Default`

Is there any highly important reason the original settings were not like this?

**Reply by osy** [reply]
Yes, they are insecure and incompatible with secure boot. If secure boot is enabled then you would have to re-personalize or reinstall macOS with a valid ApECID model. If it is disabled, the ApECID value isn't used to verify boot. But if it's set to 0 then you can never enable it in the future without having to re-personalize.

**Reply by osy** [reply]
Please resolve with OpenIntelWireless maintainers why ApECID must be set to 0 and SecureBootModel must be set to Default. These are not secure options and shouldn't be used in production environments.

**Reply by gotschi** [reply]
I had to use [Option 3] because my nuc did not show the proper USB:/OpenCore on boot, even after resetting and removing the bios jumper and resetting I only got shown "OpenCore"... no idea whats wrong with this model - the USB showed up fine on a second Hades Canyon I have just fine

**Comment by Vitoli-o** [comment]
Thank you @osy. Much appreciated!

Can I ask you the reason to support AirportItlwm over Itlwm? Could we add either option perhaps?

Most of the features of AirportItlwm are still being tested or just rolled out, whereas Itlwm is more mature. There are known issues with DHCP and Apple peripherals being disconnected when used in conjunction with the Bluetooth kext. 

On the Fan sensors, will this now allow to manage temperatures correctly when turbo boost is activated? Or is it just for reading the temperatures?

**Reply by osy** [reply]
@Malumen told me AirportItlwm was better.

**Reply by osy** [reply]
See https://github.com/osy/HaC-Mini/discussions/568

**Reply by Vitoli-o** [reply]
Thank you. I will try AirportItlwm as I updated Itlwm to v1.3.0 just over a week ago.

Re temperatures, is it worth looking at thermal profiling now that we can read the temperatures?

**Reply by osy** [reply]
I'm not sure how helpful the temperature data is since I never used it. You can try.

**Reply by Malumen** [reply]
@mr-vito  `Itlwm` is the _fallback_ for if you just need WiFi to work. It emulates a secondary `Ethernet LAN Port` and not a true WiFi like a normal mac. Some users are able to get AirDrop to work with it, and it will eventually replace `Itlwm`. I have been using `v1.2 STABLE` and `v1.3 STABLE` for months and have no problems, as well as it emulates Airport functionality. `Itlwm` needs the extra app downloaded and installed... Which if you were in a WiFi-only scenario may leave you screwed during setup. (you'd need to somehow get the `HeliPort` app onto the `HaC mini` or any other device without actually being connected to the internet). Using `AirportItlwm` allows WiFi immediately post-install (in fact, I used it during install to make a second hackintosh and was able to see nearby WiFi spots from the first-time-setup screens, very cool!).

Overall, the the `HeliPort` route is a crutch and will eventually be replaced by the `Airport` variant to keep it as seamless as possible. @osy has done a great job here with the config loading the kexts based on whichever version of macOS you are running, which theoretically would make updating to `Big Sur` a breeze.

**Reply by Needham73** [reply]
> @mr-vito `Itlwm` is the _fallback_ for if you just need WiFi to work. It emulates a secondary `Ethernet LAN Port` and not a true WiFi like a normal mac. Some users are able to get AirDrop to work with it, and it will eventually replace `Itlwm`. I have been using `v1.2 STABLE` and `v1.3 STABLE` for months and have no problems, as well as it emulates Airport functionality. `Itlwm` needs the extra app downloaded and installed... Which if you were in a WiFi-only scenario may leave you screwed during setup. (you'd need to somehow get the `HeliPort` app onto the `HaC mini` or any other device without actually being connected to the internet). Using `AirportItlwm` allows WiFi immediately post-install (in fact, I used it during install to make a second hackintosh and was able to see nearby WiFi spots from the first-time-setup screens, very cool!).
> 
> Overall, the the `HeliPort` route is a crutch and will eventually be replaced by the `Airport` variant to keep it as seamless as possible. @osy has done a great job here with the config loading the kexts based on whichever version of macOS you are running, which theoretically would make updating to `Big Sur` a breeze.


I can also add that I've been using Handoff and Universal Clipboard with AirportItlwm and it works great.

**Reply by gotschi** [reply]
> I can also add that I've been using Handoff and Universal Clipboard with AirportItlwm and it works great.

can you elaborate how you did that?
I just updated to 3.0.1, wifi and BT work great (used only BT before) but I cannot get Airdrop, Clipboard or Handoff to work

edit: I'm still on Catalina
edit2: Handoff works now, as does clipboard but Airdrop still fails

**Comment by JerryXu-code** [comment]
@osy Hello osy.macOS12 came out,and when are you going to release a new version of HaC-Mini?We are expecting for this major update!
English is not my native language,so please excuse typing errors.

**Reply by osy** [reply]
Quote from the post above

> We aim to have two major releases a year: one targeting a major update for macOS due to anticipated major changes (for example when macOS 12 comes out).

**Reply by x2358473** [reply]
HaC-Mini v3.0.1 
zh-cn:  视屏解码有问题，观看在线视频时，有时画面会卡住，但声音却在继续。
en-us:  There is a problem with video decoding. When watching online videos, sometimes the picture gets stuck, but the sound continues。

**Reply by JerryXu-code** [reply]
> Quote from the post above
> 
> > We aim to have two major releases a year: one targeting a major update for macOS due to anticipated major changes (for example when macOS 12 comes out).

Hi osy.macOS 12 came out,and you said that you will update,and we are expecting with HaC-Mini major update.When are you going to realease the new version of HaC-Mini?
English is not my native language,so please excuse typing errors.

**Reply by chrisallen** [reply]
macOS 12 is still in early beta. I would assume that support for macOS 12 will be added by the time it is officially released and stable.

**Reply by redblue9771** [reply]
@osy @chrisallen Does not provide support for Macos 12 Beta? I need to develop related applications for Beta systems. By the way, how to donate this project？

**Comment by fahdriyami** [comment]
Do I need to follow one of the options with the 3.0 release (that is marked Outdated), or should I follow them with 3.0.1? Since all of the options mention 3.0 and not 3.0.1. Im a little confused.

**Reply by Przeblysk** [reply]
3.0.1 release also need to follow one of the options

**Reply by Przeblysk** [reply]
I upgraded directly from version 2.17 to 3.0.1 release

**Comment by MattEs1969** [comment]
Thanks for all your great work, but I've hit a deadend
I went for option 3, but when I type fs0:\EFI\OC\OpenCore.efi into the shell prompt, I get "Image is not an application"
I've also tried option 4 but nothing is happening
Am I doing something wrong? Or what can I do to get it working again?

**Reply by MattEs1969** [reply]
Ok, I took out the second SSD that I had running windows, reset the BIOS and it's working again.
